tests/unit/Store/CookieStoreTest.php

```diff
@@ -48,6 +49,23 @@ public function getMock(array $args = [])
         return $mockStore;
     }
 
+    /**
+     * Gain access to PHPUnit's mock invocation stack for analyzing calls.
+     * PHPUnit 8.4 removed the native getInvocations property, requiring this workaround.
```

Comment:
what is the official phpunit alternative to this, now removed, method?

Reply:
This test was created around a private API (getInvocations) in PHPUnit, which has since been revoked. Since it was a private API, no direct alternatives were provided. With some effort this test can be rewritten to use public APIs like expects() and exactly(), but rewriting tests felt outside the scope of introducing PHP 8 support so I went this route to accommodate us in the meantime.

Now that we're moving to a modern version of PHPUnit, this will allow us to pursue a goal I've had for awhile: a complete review and restructuring of our test suite, and the introduction of tools like [Infection](https://github.com/infection/infection), [PHPStan](https://github.com/phpstan/phpstan) and [PHPBench](https://github.com/phpbench/phpbench) to give us a much more robust testing platform. This cookie test is probably my least favorite part of our current setup, so that will receive a major rework in those efforts.